modules/store/testing/src/testing.ts

```diff
@@ -18,22 +22,115 @@ export interface MockStoreConfig<T> {
   selectors?: MockSelector[];
 }
 
+/**
+ * @description
+ * Creates mock store providers.
+ *
+ * @param config `MockStoreConfig<T>` to provide the values for `INITIAL_STATE` and `MOCK_SELECTORS` tokens.
+ * By default, `initialState` and `selectors` are not defined.
+ *
+ * @usageNotes
+ *
+ * **With `TestBed.configureTestingModule`**
+ *
+ * ```ts
+ * describe('Books Component', () => {
+ *   let store: MockStore;
+ *
+ *   beforeEach(() => {
+ *     TestBed.configureTestingModule({
+ *       providers: [
+ *         provideMockStore({
+ *           initialState: { books: { entities: [] } },
+ *           selectors: [
+ *             { selector: selectAllBooks, value: ['Book 1', 'Book 2'] },
+ *             { selector: selectVisibleBooks, value: ['Book 1'] },
+ *           ],
+ *         }),
+ *       ],
+ *     });
+ *
+ *     store = TestBed.inject(MockStore);
+ *   });
+ * });
+ * ```
+ *
+ * **With `Injector.create`**
+ *
+ * ```ts
+ * describe('Counter Component', () => {
+ *   let injector: Injector;
+ *   let store: MockStore;
+ *
+ *   beforeEach(() => {
+ *     injector = Injector.create({
+ *       providers: [
+ *         provideMockStore({ initialState: { counter: 0 } }),
+ *       ],
+ *     });
+ *     store = injector.get(MockStore);
+ *   });
+ * });
+ * ```
+ */
 export function provideMockStore<T = any>(
```

Comment:
How about the following?

This is close to what you had, but I want to make sure that `useExisting` is used, and it's a bit more compacted.

```suggestion
export function provideMockStore<T = any>(
  config: MockStoreConfig<T> = {}
): Provider[] {
  setNgrxMockEnvironment(true);
  return [
    { provide: ActionsSubject, useFactory: () => new ActionsSubject(), deps: []},
    { provide: MockState, useFactory: () => new MockState<T>(), deps: []},
    { provide: MockReducerManager, useFactory: () => new MockReducerManager(), deps: []},
    { provide: INITIAL_STATE, useValue: config.initialState || {} },
    { provide: MOCK_SELECTORS, useValue: config.selectors },
    { provide: StateObservable, useExisting: MockState },
    { provide: ReducerManager, useExisting: MockReducerManager },
    {
      provide: MockStore,
      useFactory: mockStoreFactory,
      deps: [
        MockState,
        ActionsSubject,
        ReducerManager,
        INITIAL_STATE,
        MOCK_SELECTORS,
      ],
    },
    { provide: Store, useExisting: MockStore },
  ];
}
```

Reply:
@alex-okrushko 
Done. I assumed that `StateObservable` needs `useExisting: MockState`, but I added `useFactory: () => new MockState()` in order to keep the same behavior as previous `{ provide: StateObservable, useClass: MockState }`.

About return type, I kept `(ValueProvider | ExistingProvider | FactoryProvider)[]`, because if `Provider[]` is used, following code will not compile:
```typescript
injector = Injector.create({ providers: provideMockStore(...) });
``` 
because `Provider` is not compatible with `StaticProvider` and `(ValueProvider | ExistingProvider | FactoryProvider)` is compatible with both `Provider` and `StaticProvider`.